### PR TITLE
ENT-8962 Store Finality Flow peer sessions in Db for recoverability

### PR DESCRIPTION
**Pull request by colljos**

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt [hunk 1]
```diff
@@ -15,6 +19,9 @@ import net.corda.core.transactions.LedgerTransaction
 import net.corda.core.transactions.SignedTransaction
 import net.corda.core.utilities.ProgressTracker
 import net.corda.core.utilities.debug
+import net.corda.core.utilities.unwrap
+import net.corda.nodeapi.flow.FlowTransactionMetadata
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt]: I am suspicious of `core` having a dependency on `nodeapi`.  Sounds circular.  Maybe these types should just be in `core.internal` somewhere.

**Review by rick-r3 — COMMENTED** [review]
Looks about right to me, in terms of the meta it records.

File: core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt (hunk 1 above)
Comment (colljos) [reply core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt]: moved to `net.corda.core.flows `in **core** as we may wish to expose additional Flow Query operations in Enterprise.
See `FlowStatusQueryProvider`.

**Comment by colljos** [comment]
Changes folded into https://github.com/corda/corda/pull/7290

**Pull request closed without merging**